### ISSUE TITLE
Enable CORS for "app://obsidian.md"

### ISSUE THREAD

**Issue by pegasusthemis** [ollama/ollama#3827]
Hello,
I am a developer creating plugins for Obsidian, a popular knowledge management and note-taking software. I believe that enabling CORS for `app://obsidian.md` would significantly enhance the functionality and integration possibilities of Obsidian plugins with Ollama models.

Obsidian uses a custom protocol `app://obsidian.md` which I think is secure and used exclusively within the Obsidian environment. Allowing CORS for this protocol does not introduce significant security risks as it is a controlled environment, primarily dealing with local files and user-initiated actions. This environment ensures that enabling CORS would not expose Ollama to unwanted cross-origin requests in a way that compromises security.

By enabling CORS, developers can seamlessly integrate Ollama's advanced modeling capabilities directly into Obsidian, enhancing the user experience and providing advanced features directly within the application. 

I dont know if I express myself properly, but I just ask, if it would be possible to put app://obsidian.md in OLLAMA_ORIGINS to avoid using * in this case.

Thank you for considering this enhancement. I appreciate the fantastic work you've done with Ollama and fully respect any decision you make regarding this request. Your efforts in developing such a powerful tool are greatly appreciated by the community.

Best regards,

Pegasus Themis

**Comment by TfTHacker** [comment]
@pegasusthemis I am curious if you got this working on Mac? this looks like the linux way to set the env variable.

**Comment by ransurf** [comment]
haha fun seeing you here :) is this for development or are you using a
program? i think when i used lm studio i had an option to disable cors in
server settings and it fixed the issue? i do not fully remember though

**Comment by pegasusthemis** [comment]
TfTHacker: Nope I configured this on a Windows 11 machine using the Environment Variables GUI. For setting it up across multiple Windows systems, I employed the command line as admin, with the following syntax: SETX /M OLLAMA_ORIGINS "app://obsidian.md*" which applies the setting system-wide.

I've noticed that you are also developing plugins for Obsidian, with AI. I'm excited to try out your plugin and see how it does. Best luck with your development. Looking forward to see more of your work!

Ransurf: I develop a plugin for Obsidian, already fix the CORS problems with Ollama, but I am having a little bit of a headache with the Claude API CORS

Warm regards,

Pegasus Themis #
